Terminal.Gui/View/Layout/Dim.cs

```diff
@@ -0,0 +1,766 @@
+    /// <summary>
+    ///     Gets the right dimension.
+    /// </summary>
+    public Dim Right { get; } = right;
+
+    /// <inheritdoc/>
+    public override string ToString () { return $"Combine({Left}{(Add ? '+' : '-')}{Right})"; }
+
+    internal override int Anchor (int size)
+    {
+        int la = Left.Anchor (size);
+        int ra = Right.Anchor (size);
+
+        if (Add)
+        {
+            return la + ra;
+        }
+
+        return la - ra;
+    }
+
+    internal override int Calculate (int location, int superviewContentSize, View us, Dimension dimension)
+    {
+        int leftNewDim = Left.Calculate (location, superviewContentSize, us, dimension);
+        int rightNewDim = Right.Calculate (location, superviewContentSize, us, dimension);
+
+        int newDimension;
+
+        if (Add)
+        {
+            newDimension = leftNewDim + rightNewDim;
+        }
+        else
+        {
+            newDimension = Math.Max (0, leftNewDim - rightNewDim);
+        }
+
+        return newDimension;
+    }
+
+    /// <summary>
+    ///     Diagnostics API to determine if this Dim object references other views.
+    /// </summary>
+    /// <returns></returns>
+    internal override bool ReferencesOtherViews ()
+    {
+        if (Left.ReferencesOtherViews ())
+        {
+            return true;
+        }
+
+        if (Right.ReferencesOtherViews ())
+        {
+            return true;
+        }
+
+        return false;
+    }
+}
+
+/// <summary>
+///     Represents a dimension that is a percentage of the width or height of the SuperView.
+/// </summary>
+/// <remarks>
+///     This is a low-level API that is typically used internally by the layout system. Use the various static
+///     methods on the <see cref="Dim"/> class to create <see cref="Dim"/> objects instead.
+/// </remarks>
+/// <param name="percent">The percentage.</param>
+/// <param name="usePosition">
+///     If <see langword="true"/> the dimension is computed using the View's position (<see cref="View.X"/> or
+///     <see cref="View.Y"/>).
+///     If <see langword="false"/> the dimension is computed using the View's <see cref="View.ContentSize"/>.
+/// </param>
+public class DimPercent (float percent, bool usePosition = false) : Dim
+{
+    /// <inheritdoc/>
+    public override bool Equals (object other) { return other is DimPercent f && f.Percent == Percent && f.UsePosition == UsePosition; }
+
+    /// <inheritdoc/>
+    public override int GetHashCode () { return Percent.GetHashCode (); }
+
+    /// <summary>
+    ///     Gets the percentage.
+    /// </summary>
+    public new float Percent { get; } = percent;
+
+    /// <summary>
+    /// </summary>
+    /// <returns></returns>
+    public override string ToString () { return $"Percent({Percent},{UsePosition})"; }
+
+    /// <summary>
+    ///     Gets whether the dimension is computed using the View's position or ContentSize.
+    /// </summary>
+    public bool UsePosition { get; } = usePosition;
```

Comment:
Another one that would be much better as an enum, with the property's name adjusted accordingly.

For example, `PositionReference` or `PositionMode` or whatever, and the enum having the possible behaviors.